modules/phase_field/src/kernels/ACGrGrPoly.C

```diff
@@ -18,20 +18,15 @@ InputParameters validParams<ACGrGrPoly>()
 
 ACGrGrPoly::ACGrGrPoly(const InputParameters & parameters) :
     ACBulk(parameters),
+    _ncrys(coupledComponents("v")),
+    _vals(_ncrys),
+    _vals_var(_ncrys),
     _mu(getMaterialProperty<Real>("mu")),
     _gamma(getMaterialProperty<Real>("gamma_asymm")),
     _tgrad_corr_mult(getMaterialProperty<Real>("tgrad_corr_mult")),
-    _has_T(isCoupled("T")),
```

Comment:
The reason behind the "has_T" thing is that there is an issue with the grain growth model that we never figured out. When you run a grain growth simulation with a coupled temperature that has a gradient, the GBs migrate magically down the temperature gradient. This is different than the actual temperature gradient driving force that causes GBs to move up the temperature gradient and is added using ACGrGrTGrad in MARMOT. Since I couldn't figure out what was causing it, I tried adding a correction driving force that resisted the incorrect motion. It didn't really fix the problem so it is fine to take it out. However, I do want to make sure you are aware of the problem.